virtio-devices/src/vsock/packet.rs

```diff
@@ -153,9 +166,19 @@ impl VsockPacket {
             return Err(VsockError::BufDescTooSmall);
         }
 
+        let buf_desc_addr = if let Some(access_platform) = access_platform {
+            GuestAddress(
+                access_platform
+                    .translate(buf_desc.addr().0, u64::from(buf_desc.len()))
+                    .unwrap(),
+            )
+        } else {
+            buf_desc.addr()
+        };
+
```

Comment:
Could you do something like this

```rust
trait Translateable {
   fn translate(&self,access_platform: Option<&Arc<dyn AccessPlatform>>, len: usize) -> Self;
}

impl Translatable for GuestAddress 
{
   fn translate(&self,access_platform: Option<&Arc<dyn AccessPlatform>>, len: usize) -> Self {
        if let Some(access_platform) = access_platform {
            GuestAddress(
                access_platform
                    .translate(self.0, u64::from(len))
                    .unwrap(),
            )
        } else {
           self
        }
    }
}


```
And then just use `.translate()` instead of this every time. You can add new traits to existing structs if you do it in the crate where the trait is defined. So you could just put this in the virtio-devices crate.

Reply:
Nice suggestion, I've implemented it.